### ISSUE TITLE
Impostor cylinders issue

### ISSUE THREAD

**Issue by giagitom** [molstar/molstar#831]
I noticed an issue on impostor cylinders, here is a state to reproduce it [test.molj](https://github.com/molstar/molstar/files/11533051/test.molj.txt)

![image](https://github.com/molstar/molstar/assets/20627977/665227e2-7132-45a7-9a4c-e0a635edb0e9)

**Comment by arose** [comment]
That is quite an edge case. As far as I can tell it only happens when the distance between near and far plane is about the size of a cylinder. Could you just disallow bringing near and far plane that close together?

**Comment by giagitom** [comment]
I first noticed this while zooming in using middle mouse click+drag default event. Maybe I could try to use a custom event for this. For longer cylinders the effect seems to be more visible though. Here an example with HEME [test-heme.molj](https://github.com/molstar/molstar/files/11544176/test.molj.txt)
![image](https://github.com/molstar/molstar/assets/20627977/a0fff0da-3069-4ebf-903a-849a783b20fe)

**Comment by giagitom** [comment]
@arose I think I have found the cause in this line in vertex shader

https://github.com/molstar/molstar/blob/3bf4a8f8e692dd0f2159e9734e922d334aed1bdc/src/mol-gl/shader/cylinders.vert.ts#L75

Modifying it in the way below it seems to work fine. I don't know if there are drawbacks on doing so though. Do you think this would be an acceptable fix?

```mvPosition.z -= length(vEnd - vStart) + vSize; // avoid clipping```

![image](https://github.com/molstar/molstar/assets/20627977/077529d5-04a2-4571-867c-607ae99233a7)

**Comment by arose** [comment]
Right, thanks, that should be `mvPosition.z -= length(vEnd - vStart) + 2.0 * vSize; // avoid clipping` since vSize is the cylinder radius.
